x-pack/plugins/security_solution/public/management/hooks/endpoint/use_get_endpoints_list.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UseQueryOptions, UseQueryResult } from '@tanstack/react-query';
+import { useQuery } from '@tanstack/react-query';
+import type { IHttpFetchError } from '@kbn/core-http-browser';
+import { useHttp } from '../../../common/lib/kibana';
+import type { HostInfo, MetadataListResponse } from '../../../../common/endpoint/types';
+import { HOST_METADATA_LIST_ROUTE } from '../../../../common/endpoint/constants';
+
+type GetEndpointsListResponse = Array<{
+  id: HostInfo['metadata']['agent']['id'];
+  name: HostInfo['metadata']['host']['hostname'];
+  selected: boolean;
+}>;
+
+const PAGING_PARAMS = Object.freeze({
+  default: 50,
+  all: 10000,
+});
+/**
+ * Get info for a security solution endpoint host using hostnames via kuery param
+ * page and pageSize are fixed for showing 50 hosts at most on the 1st page
+ * @param query
+ * @param options
+ */
+export const useGetEndpointsList = ({
+  searchString,
+  selectedAgentIds,
+  options = {},
+}: {
+  searchString: string;
+  selectedAgentIds?: string[];
+  options?: UseQueryOptions<GetEndpointsListResponse, IHttpFetchError>;
+}): UseQueryResult<GetEndpointsListResponse, IHttpFetchError> => {
+  const http = useHttp();
+  const kuery = `united.endpoint.host.hostname:${searchString.length ? `*${searchString}` : ''}*`;
```

Comment:
there are some `kueries` here and in the test file ^^

Reply:
Not sure what you mean.

Reply:
Oh, I believed it was a typo for query, sorry. A quick search for the word `kuery` didn't help during the review - now I searched more thoroughly to find it's related to KQL. 😅

Reply:
Yeah, I wondered the same when I saw `kuery` (why spell `query` with a `k`) during my first couple of weeks at Elastic. 🤣